apps/teams-test-app/src/components/AuthenticationAPIs.tsx

```diff
@@ -17,19 +18,55 @@ const GetAuthToken = (): React.ReactElement =>
   ApiWithTextInput<authentication.AuthTokenRequestParameters>({
     name: 'getAuthToken',
     title: 'Get Auth Token',
-    onClick: async authParams => {
-      const result = await authentication.getAuthToken(authParams);
-      return 'Success: ' + JSON.stringify(result);
+    onClick: {
+      validateInput: () => {
+        return; //This API can have no input
+      },
+      submit: {
+        withPromise: async authParams => {
+          const params = authParams;
+          const result = await authentication.getAuthToken(params);
+          return 'Success: ' + JSON.stringify(result);
+        },
+        withCallback: (authParams, setResult) => {
+          const callback = (result: string): void => {
+            setResult(result);
+          };
+          const authRequest: authentication.AuthTokenRequest = {
+            successCallback: callback,
+            failureCallback: callback,
+            ...authParams,
+          };
+          authentication.getAuthToken(authRequest);
+          return 'getAuthToken()' + noHostSdkMsg;
+        },
+      },
     },
   });
 
 const GetUser = (): React.ReactElement =>
   ApiWithoutInput({
     name: 'getUser',
     title: 'Get User',
-    onClick: async () => {
-      const user = await authentication.getUser();
-      return 'Success: ' + JSON.stringify(user);
+    onClick: {
+      withPromise: async () => {
+        const user = await authentication.getUser();
+        return 'Success: ' + JSON.stringify(user);
+      },
+      withCallback: setResult => {
+        const successCallback = (user: authentication.UserProfile): void => {
+          setResult(JSON.stringify(user));
```

Comment:
I should probably make this consistent with the Promise call by either removing the 'Success' in the withPromise call or adding the 'Success' to this call, I am indifferent to which approach we go with and want to know others feedback

Reply:
Looks like there's no E2E scenario test around this API call, so we should be fine with either. I would probably drop the `success: ` from both.

Reply:
Is this also an inconsistency that we should clean up in GetAuthToken?